listenbrainz/spark/request_manage.py

```diff
@@ -0,0 +1,92 @@
+import click
+import listenbrainz.utils as utils
+import os
+import pika
+import ujson
+
+from flask import current_app
+from listenbrainz.webserver import create_app
+
+
+QUERIES_JSON_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'request_queries.json')
+
+cli = click.Group()
+
+class InvalidSparkRequestError(Exception):
+    pass
+
+
+def _get_possible_queries():
+    """ Return the dict describing all possible queries that can
+    be sent to Spark. Listed in listenbrainz/spark/request_queries.json
+    """
+    with open(QUERIES_JSON_PATH) as f:
+        return ujson.load(f)
+
+
+def _prepare_query_message(query, params=None):
+    """ Prepare the JSON message that needs to be sent to the
+    spark cluster based on the query and the parameters the
+    query needs
+
+    Args:
+        query (str): the name of the query, should be in request_queries.json
+        params (dict): the parameters the query needs, should contain all the params
+            in the correspoding request_queries.json to be valid
+
+    Raises:
+        InvalidSparkRequestError if the query isn't in the list or if the parameters
+        don't match up
+    """
+    if params is None:
+        params = {}
+
+    possible_queries = _get_possible_queries()
+    if query not in possible_queries:
+        raise InvalidSparkRequestError(query)
+
+    message = {'query': possible_queries[query]['name']}
+
+    required_params = set(possible_queries[query]['params'])
+    given_params = set(params.keys())
+    if required_params != given_params:
+        raise InvalidSparkRequestError
+
+    if params:
+        message['params'] = {}
+    for key, value in params.items():
```

Comment:
I'd indent this block as well, less to do, but we'll never notice a difference in real life...